roadmap/README.md

```diff
@@ -0,0 +1,149 @@
+  - [Catalog](./catalog.md)
+
+## Mission and Vision
+
+Tekton's mission:
+
+  Be the industry-standard, cloud-native CI/CD platform components and ecosystem.
+
+The vision for this is:
+
+* Tekton API conformance across as many CI/CD platforms as possible
+* A rich catalog of high quality, reusable `Tasks` which work with Tekton conformant systems
+
+What this vision looks like differs across different users:
+
+* **Engineers building CI/CD systems**: These users will be motivated to use Tekton
+  and integrate it into the CI/CD systems they are using because building on top
+  of Tekton means they don't have to re-invent the wheel and out of the box they get
+  scalable, serverless cloud native execution
+* **Engineers who need CI/CD**: (aka all software engineers!) These users will benefit
+  from the rich high quality catalog of reusable components:
+
+  * Quickly build and interact with sophisticated `Pipelines`
+  * Be able to port `Pipelines` to any Tekton conformant system
+  * Be able to use multiple Tekton conformant systems instead of being locked into one
+    or being forced to build glue between multiple completely different systems
+  * Use an ecosystem of tools that know how to interact with Tekton components, e.g.
+    IDE integrations, linting, CLIs, security and policy systems
+
+## Roadmap
+
+### Beta and GA
+
+In 2019 we got to the point where we had several projects built on top of Tekton
+that were craving additional API stability, so we started a push toward having
+a beta release of Tekton. We have
+[created a plan](https://docs.google.com/document/d/1H8I2Rk4kLdQaR4mV0A71Qbk-1FxXFrmvisEAjLKT6H0/edit)
+which defines what Beta means for Tekton Pipelines, and
+[through our beta working group](https://github.com/tektoncd/community/blob/master/working-groups.md#beta-release)
+we are working towared a beta release in early 2020.
+
+The initial beta release
+[does not include all Tekton Pipeline components](https://docs.google.com/document/d/1H8I2Rk4kLdQaR4mV0A71Qbk-1FxXFrmvisEAjLKT6H0/edit#heading=h.t0sc4hdrr5yq). After our initial
+beta release, we will then work toward:
+
+1. Beta for the remaining components
+1. Beta for [Tekton Triggers](#triggers)
+1. GA for both Tekton Pipelines and Tekton Triggers
+
+As the project matures, we also require:
+
+1. A website that provides a good landing page for users (tekton.dev)
+1. Solid, high quality onboarding and documentation
+
+### "Feature Complete" Tekton
+
+Today Tekton can do basically anything you want it to, but there are a few features
+still missing or not quite where we want them. Once we have these features, folks
+should be able to use Tekton for all of their CI/CD use cases.
+
+Deciding what to support and what to expect users to write into their own `Tasks`
+is a constant balancing act, and we want to be careful and deliberate with what we
+include.
+
+Features we don't have or aren't yet 100% satisfied with for "feature complete"
+Tekton:
```

Comment:
Other ideas:
- Looping: https://github.com/tektoncd/pipeline/issues/2050, https://github.com/tektoncd/pipeline/issues/2112
- Alternative Task types - e.g. tasks with no underlying pod - for instance when waiting on input for an arbitrarily long time or so.